wagtail/images/models.py

```diff
@@ -468,6 +469,24 @@ def get_rendition(self, filter: Union["Filter", str]) -> "AbstractRendition":
 
         if isinstance(filter, str):
             filter = Filter(spec=filter)
+        elif not isinstance(filter, Filter):
+            raise InvalidFilterSpecError(
+                "Unrecognised filter format - string or Filter instance expected"
+            )
+
+        if "preserve-svg" in filter.spec:
+            # remove 'preserve-svg' from filter specs for all image types
+            filter.spec = "|".join(
+                item for item in filter.spec.split("|") if item != "preserve-svg"
+            )
+            if not filter.spec:
+                # no formatting directives were included in filter
+                raise InvalidFilterSpecError(
+                    "Filter should include at least one formatting directive other than 'preserve-svg'"
+                )
+            if self.is_svg():
+                # remove rasterizing directives
+                filter.spec = to_svg_safe_spec(filter.spec)
```

Comment:
If we were to proceed with this approach, this would need implementing in `get_renditions` as well.